cpp/src/plane_extraction.cpp

```diff
@@ -0,0 +1,569 @@
+#include <numeric>
+#include "deplex/plane_extraction.h"
+
+#ifdef DEBUG_CAPE
+#include <iostream>
+#include <fstream>
+#endif
+
+namespace deplex {
+
+const config::Config CAPE::__default_config{
+    {// General parameters
+     {"patchSize", "12"},
+     {"histogramBinsPerCoord", "20"},
+     {"minCosAngleForMerge", "0.93"},
+     {"maxMergeDist", "500"},
+     {"minRegionGrowingCandidateSize", "5"},
+     {"minRegionGrowingCellsActivated", "4"},
+     {"minRegionPlanarityScore", "50"},
+     {"doRefinement", "true"},
+     {"refinementMultiplierCoeff", "15"},
+     // Parameters used in plane validation
+     {"depthSigmaCoeff", "1.425e-6"},
+     {"depthSigmaMargin", "10"},
+     {"minPtsPerCell", "3"},
+     {"depthDiscontinuityThreshold", "160"},
+     {"maxNumberDepthDiscontinuity", "1"}}};
+
+CAPE::CAPE(int32_t image_height, int32_t image_width, config::Config config)
+    : _config(config),
+      _nr_horizontal_cells(image_width / config.getInt("patchSize")),
```

Comment:
Maybe better to keep patch_size as class variable? not to access it from config everytime, it is oftenly used